go.mod

```diff
@@ -69,7 +69,7 @@ require (
 	github.com/mitchellh/gox v1.0.1
 	github.com/mitchellh/reflectwalk v1.0.2
 	github.com/nishanths/exhaustive v0.7.11
-	github.com/openbao/openbao/api v0.0.0-20240326035453-c075f0ef2c7e
+	github.com/openbao/openbao/api/v2 v2.1.0
```

Comment:
To sync some discussions with @cam72cam -- I went back to `v2.1.0` since `v2.2.0` bumped `hcl` to `v1.0.1-vault-5` tag, which I know is probably potentially breaking for you all more so than anything you get out of the `v2.2.0` tag. 

I will note the `v2.2.0` tag has some fixes for Go 1.24, but not sure if that'll impact you or not. Worst case you can audit the `hcl` v1 bump later if Go 1.24 causes problems and bump OpenBao dep again then.